get history power data paginated

leneffets opened the pull request:
Hey there,

based on the issue https://github.com/suaveolent/hoymiles-wifi/issues/15 I modified the history data functino, so that more datapoints will be recieved. it takes some time, as it makes more requests, but for me on hoymiles HMS-800W-2T it works fine.

Thanks
leneffets

suaveolent commented:
Thanks for the addition.


Upon quick review, I saw that using AI to explain all the paramters was probably not a good idea from my side.

We might need to change `control_point` back to `cp` (I assume it means `current_page` and `power_array` back to `ap` (which most likely means `all_pages`).

Could you maybe do that? 

Also could you maybe have a look at how pagination is implemented in  ` async_get_real_data_new` ? I would prefer a similar approach to avoid different implementations for the same feature.

leneffets commented:
Okay, i will check that.

suaveolent ***@***.***> schrieb am Mi., 2. Apr. 2025, 14:14:

> Thanks for the addition.
>
> Upon quick review, I saw that using AI to explain all the paramters was
> probably not a good idea from my side.
>
> We might need to change control_point back to cp (I assume it means
> current_page and power_array back to ap (which most likely means all_pages
> ).
>
> Could you maybe do that?
>
> Also could you maybe have a look at how pagination is implemented in
> async_get_real_data_new ? I would prefer a similar approach to avoid
> different implementations for the same feature.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/suaveolent/hoymiles-wifi/pull/57#issuecomment-2772372235>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AR3TM5PPKGR37RGWZ7EYZFT2XPIBVAVCNFSM6AAAAAB2G25DOCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZSGM3TEMRTGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: suaveolent]*suaveolent* left a comment
> (suaveolent/hoymiles-wifi#57)
> <https://github.com/suaveolent/hoymiles-wifi/pull/57#issuecomment-2772372235>
>
> Thanks for the addition.
>
> Upon quick review, I saw that using AI to explain all the paramters was
> probably not a good idea from my side.
>
> We might need to change control_point back to cp (I assume it means
> current_page and power_array back to ap (which most likely means all_pages
> ).
>
> Could you maybe do that?
>
> Also could you maybe have a look at how pagination is implemented in
> async_get_real_data_new ? I would prefer a similar approach to avoid
> different implementations for the same feature.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/suaveolent/hoymiles-wifi/pull/57#issuecomment-2772372235>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AR3TM5PPKGR37RGWZ7EYZFT2XPIBVAVCNFSM6AAAAAB2G25DOCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZSGM3TEMRTGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

leneffets commented:
Hey there, 
* i now i modifed the PR and updated the protoc files according to your information.
* i also modified the devcontainers. so  that it runs in my local network, can test directly agains my hoymiles.
* i adapted same mechanism for getting history data.

leneffets commented:
@suaveolent any news on that topic?

suaveolent commented:
@leneffets I finally managed to implement the hybrid inverter command. This however, also results in some merge conflicts. I will cherry pick your PR.
